src/client/clientmedia.cpp

```diff
@@ -254,6 +254,10 @@ void ClientMediaDownloader::initialStep(Client *client)
 			fetch_request.post_data = required_hash_set;
 			fetch_request.extra_headers.emplace_back(
 				"Content-Type: application/octet-stream");
+			fetch_request.extra_headers.emplace_back(
+				std::string("Referer: minetest://") +
+				client->getAddressName() + ":" +
```

Comment:
this looks like it wouldn't account for a raw IPv6 which needs to be enclosed in `[]` to form a valid URI

Reply:
do people do that? lol

I suppose I can use the methods to determine if it is ipv6, but testing if it's raw would, I think, mean parsing the actual address string if I'm not mistaken.

I'll look at the Address class methods...

Reply:
eh probably testing for `:` would be sufficient... I'll see if i can cook that up.

Reply:
Please check this, I think it'll be ok.